$NetBSD: patch-ab,v 1.13 2016/11/28 14:40:42 leot Exp $

o Handle the various operating systems in the same way avoiding hardcoding
o Properly recognize glfw support

--- Makerules.orig	2016-11-21 10:21:11.000000000 +0000
+++ Makerules
@@ -60,51 +60,6 @@ ifeq "$(largefile)" "yes"
 CFLAGS += -DFZ_LARGEFILE
 endif
 
-# Windows (MINGW) build doesn't use system libraries.
-ifeq "$(OS)" "MINGW"
-
-WINDRES = windres
-HAVE_X11 ?= no
-HAVE_WIN32 = yes
-CC=gcc
-# Add flags required for Harfbuzz because of missing strdup
-CFLAGS+= -U__STRICT_ANSI__
-
-# Mac OS X doesn't have pkg-config so we hard code paths.
-else ifeq "$(OS)" "MACOS"
-
-HAVE_X11 ?= no
-HAVE_PTHREADS ?= yes
-
-# Mac OS X deprecated libcrypto, so the default is to not include it.
-HAVE_LIBCRYTO ?= no
-SYS_LIBCRYPTO_CFLAGS =
-SYS_LIBCRYPTO_LIBS =
-ifeq "$(HAVE_LIBCRYPTO)" "yes"
-SYS_LIBCRYPTO_CFLAGS = -DHAVE_LIBCRYPTO
-SYS_LIBCRYPTO_LIBS = -lcrypto
-endif
-
-SYS_CURL_DEPS = -lpthread
-
-SYS_X11_CFLAGS = -I/usr/X11R6/include
-SYS_X11_LIBS = -L/usr/X11R6/lib -lX11 -lXext
-
-SYS_FREETYPE_CFLAGS = $(shell freetype-config --cflags)
-SYS_FREETYPE_LIBS = $(shell freetype-config --libs)
-SYS_OPENJPEG_LIBS = -lopenjpeg
-SYS_JBIG2DEC_LIBS = -ljbig2dec
-SYS_JPEG_LIBS = -ljpeg
-SYS_ZLIB_LIBS = -lz
-
-CC = xcrun cc
-AR = xcrun ar
-LD = xcrun ld
-RANLIB_CMD = xcrun ranlib $@
-
-# Linux uses pkg-config for system libraries.
-else ifeq "$(OS)" "Linux"
-
 HAVE_PTHREADS ?= yes
 
 ifeq "$(shell pkg-config --exists libcrypto && echo yes)" "yes"
@@ -131,6 +86,17 @@ SYS_X11_CFLAGS = $(shell pkg-config --cf
 SYS_X11_LIBS = $(shell pkg-config --libs x11 xext)
 endif
 
+ifeq "$(shell pkg-config --exists glut && echo yes)" "yes"
+SYS_GLUT_CFLAGS = $(shell pkg-config --cflags glut)
+SYS_GLUT_LIBS = $(shell pkg-config --libs glut)
+endif
+
+ifeq "$(shell pkg-config --exists glfw3 && echo yes)" "yes"
+HAVE_GLFW = yes
+SYS_GLFW_CFLAGS = $(shell pkg-config --cflags glfw3)
+SYS_GLFW_LIBS = $(shell pkg-config --libs glfw3)
+endif
+
 ifeq "$(shell pkg-config --exists harfbuzz && echo yes)" "yes"
 SYS_HARFBUZZ_CFLAGS = $(shell pkg-config --cflags harfbuzz)
 SYS_HARFBUZZ_LIBS = $(shell pkg-config --libs harfbuzz)
@@ -150,8 +116,6 @@ SYS_JBIG2DEC_LIBS = -ljbig2dec
 SYS_JPEG_LIBS = -ljpeg
 SYS_ZLIB_LIBS = -lz
 
-endif
-
 # The following section is an example of how to simply do cross-compilation
 # using these Makefiles. It builds for a beagleboard running ARM linux,
 # compiling on windows with the CodeSourcery G++ compilers.
